### PR TITLE
Make cellCameraInit returns CELL_OK when no camera is plugged in.

### PR DESCRIPTION
**Pull request by scribam**
Tested on DECR.

- [X] Rapala Pro Bass Fishing (Nothing => Ingame)

screenshot from 2018-03-05 18-34-07

screenshot from 2018-03-05 18-44-12

### CODE REVIEW

**Comment by VelocityRa** [comment]
What's the point of this? Why should it report that it's there if the Camera Handler is Null?
Isn't the same effect achieved with Camera Handler set to Fake?

**Comment by Megamouse** [comment]
I think the whole point of null is to fake things

**Comment by VelocityRa** [comment]
As opposed to "Fake" being there to fake things? 🤔

**Comment by scribam** [comment]
If you set "Camera Handler" to "Fake", this game goes also ingame, likely thanks to your previous PR.
Maybe I am mistaking something but the option "Fake" means to me we are simulating a connected camera. What I want to address in this PR is when you have no camera connected.

**Comment by VelocityRa** [comment]
I think you misunderstood what Fake is. You just set it on the dropdown Input/Output settings, you don't need any hardware.

Null should emulate no camera, Fake should emulate a (fake) camera. That way all bases are covered.

**Comment by scribam** [comment]
I haven't expressed myself clearly then.
The attempt of this PR is to improve the "Null" emulation. HW tests on the DECR show that cellCameraInit returns CELL_OK when no camera is connected.

**Comment by VelocityRa** [comment]
Ah ok so you tested on h/w, didn't get that.
Just verified that cameraInit can never return `CELL_CAMERA_ERROR_DEVICE_NOT_FOUND` by looking at libGem as well.

**Comment by AniLeo** [comment]
So, without camera, the camera thread is not initiated, it just returns CELL_OK?

**Comment by scribam** [comment]
Well, I don't have the answer to this question. @VelocityRa, do you know anything about the camera thread?

**Comment by VelocityRa** [comment]
Yes, I made the camera thread. Its purpose is to send messages to the event queue registered with the `cellCameraSetNotifyEventQueue*` functions. Such messages are camera connected, camera disconnected and frame update events. It is therefore useless if the camera handler is set to Null, so the current code is fine.

**Comment by AniLeo** [comment]
If PS3 starts thread regardless it can affect behavior, because we'll be returning CELL_OK instead of ALREADY_INIT.

**Comment by VelocityRa** [comment]
@AniLeo That's true yeah.
Maybe the `if` below
```cpp
const auto g_camera = fxm::make<camera_thread>();
```
and be done with it.

**Comment by AniLeo** [comment]
But we need to confirm if it ever starts the thread if camera is not connected in the first place

**Comment by VelocityRa** [comment]
The enum being called `ALREADY_INIT` makes the most sense to just mean "`cellCameraInit` has been called before without `cellCameraEnd` being called" to me, especially since it returning `DEVICE_NOT_FOUND` is out of the question, but if it's easy for @scribam to test, sure.

**Comment by scribam** [comment]
Confirmed, a PS3 without camera doesn't start a thread.

File: rpcs3/Emu/Cell/Modules/cellCamera.cpp [hunk 1]
```diff
@@ -359,6 +379,11 @@ s32 cellCameraIsAvailable(s32 dev_num)
 {
 	cellCamera.todo("cellCameraIsAvailable(dev_num=%d)", dev_num);
 
+	if (g_cfg.io.camera == camera_handler::null)
+	{
+		return false;
```
Comment (AniLeo) [inline rpcs3/Emu/Cell/Modules/cellCamera.cpp]: Why not CELL_OK? They both represent 0

Comment (VelocityRa) [reply rpcs3/Emu/Cell/Modules/cellCamera.cpp]: Why? This is more correct semantically. You're relying on the fact that `CELL_OK == 0` for no reason.
Also it's consistent with the rest of the file.

**Comment by AniLeo** [comment]
This has nothing to do with rendering, this is a cellCamera pull request. Don't flood

**Pull request merged**